### PR TITLE
Enhance `NumberFormatOptions.colorSpec` to accept custom CSS properti…

### PR DESCRIPTION
**Pull request by ghsolomon**
…es in addition to class names

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Review by haynesjm42 — APPROVED** [review]
Looks good to me, will leave this open for @amcclain or @lbwexler to review/comment given the controversial naming of this thing :)

File: format/FormatNumber.ts [hunk 1]
```diff
@@ -393,13 +398,20 @@ function signGlyph(v: number, asHtml: boolean = false) {
...
-    if (v < 0) return colorSpec.neg;
-    if (v > 0) return colorSpec.pos;
-    return colorSpec.neutral;
+
+    const possibleClassName = v < 0 ? colorSpec.neg : v > 0 ? colorSpec.pos : colorSpec.neutral;
+    return isString(possibleClassName) ? possibleClassName : '';
+}
+
+function valueStyle(v: number, colorSpec: ColorSpec | boolean): CSSProperties {
+    if (!isFinite(v) || isBoolean(colorSpec)) return {};
```
Comment (lbwexler) [inline format/FormatNumber.ts:411]: why the assymetry between line 402 and 411?

**Comment by lbwexler** [comment]
My only problem with the names are actually the internal names `valueClass` and `valueStyle`.  How about something like `classNameFromColorSpec`  and `stylesFromColorSpec`

**Comment by lbwexler** [comment]
Also, can you please remind me why this is needed by applications?  (versus requiring them to create classes to house the styles)

**Pull request merged**